### PR TITLE
feat: add C implementation for `math/base/special/asech`

### LINKED ISSUE #862: [RFC]: Add C implementation for `@stdlib/math/base/special/asech`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/asech`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/asech/).

```c
double stdlib_math_asech( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Userf1995**
Resolves #862 .

## Description

> What is the purpose of this pull request?

This pull request: Add C implementation for @stdlib/math/base/special/asech

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #862 
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [X] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

### CODE REVIEW

**Comment by Userf1995** [comment]
@Pranavchiku, @Planeshifter please review these PR,and let me know if any changes need to be made.

File: lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/benchmark.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = rand_double();
+		y = acosh( 1.0/x );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/benchmark.c]:
```suggestion
		y = acosh( 1.0 / x );
```

File: lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/native/Makefile [hunk 2]
```diff
@@ -0,0 +1,146 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/native/Makefile]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/native/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,136 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/benchmark/c/native/benchmark.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/binding.gyp [hunk 4]
```diff
@@ -0,0 +1,170 @@
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/binding.gyp]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/examples/c/Makefile [hunk 5]
```diff
@@ -0,0 +1,146 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/examples/c/Makefile]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/examples/c/example.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/examples/c/example.c]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+int main( void ) {
+    const double x[] = { 0.0, 0.5, 1.0, 1.89, 2.33, 3.22, 3.67, 4.11, 4.56, 5.0 };
+
+    double v;
+    int i;
+    for ( i = 0; i < 10; i++ ) {
+        v = stdlib_base_asech( x[ i ] );
+        printf( "asech(%lf) = %lf\n", x[ i ], v );
+    }
+}
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/examples/c/example.c]:
```suggestion
int main( void ) {
	const double x[] = { 0.0, 0.5, 1.0, 1.89, 2.33, 3.22, 3.67, 4.11, 4.56, 5.0 };

	double v;
	int i;
	for ( i = 0; i < 10; i++ ) {
		v = stdlib_base_asech( x[ i ] );
		printf( "asech(%lf) = %lf\n", x[ i ], v );
	}
}
```

File: lib/node_modules/@stdlib/math/base/special/asech/include.gypi [hunk 8]
```diff
@@ -0,0 +1,53 @@
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/include.gypi]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/include/stdlib/math/base/special/asech.h [hunk 9]
```diff
@@ -0,0 +1,38 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/include/stdlib/math/base/special/asech.h]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/include/stdlib/math/base/special/asech.h [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Computes the hyperbolic arcsecant of x.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/include/stdlib/math/base/special/asech.h]:
```suggestion
* Computes the hyperbolic arcsecant of double-precision floating-point number.
```

File: lib/node_modules/@stdlib/math/base/special/asech/lib/main.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2022 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/lib/main.js]:
```suggestion
* Copyright (c) 2022 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/asech/src/main.c [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+*/
+
+#include "stdlib/math/base/special/asech.h"
+#include "stdlib/math/base/special/acosh.h"
+
+/**
+* Compute the hyperbolic arcsecant of a number.
+*
+* @param x    input value
+* @return	  output value
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/src/main.c]:
```suggestion
* @param x    input value
* @return     output value
```

File: lib/node_modules/@stdlib/math/base/special/asech/src/main.c [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+*
+* @example
+* out = stdlib_base_asech( 1.0 );
+* // returns 0.0
+*/
+double stdlib_base_asech( const double x ) {
+    return stdlib_base_acosh( 1.0 / x );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/asech/src/main.c]:
```suggestion
	return stdlib_base_acosh( 1.0 / x );
```

**Review by Pranavchiku — APPROVED** [review]
Tested locally, everything looks good to me. Thanks @Rejoan-Sardar !

File: lib/node_modules/@stdlib/math/base/special/asech/lib/native.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Compute the hyperbolic arcsecant of a number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/asech/lib/native.js]:
```suggestion
* Computes the hyperbolic arcsecant of a number.
```

File: lib/node_modules/@stdlib/math/base/special/asech/src/main.c [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/asech.h"
+#include "stdlib/math/base/special/acosh.h"
+
+/**
+* Compute the hyperbolic arcsecant of a number.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/asech/src/main.c]:
```suggestion
* Computes the hyperbolic arcsecant of a number.
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**